sip/sip-001-burn-election.md

```diff
@@ -0,0 +1,1074 @@
+selfish mining attacks, this gives honest leaders the ability to identify and reject an attack fork 
+with over 50% confidence.  In particular, honest leaders who have been online long
+enough to measure the expected block propagation time would _not_ build on top of
+a chain tip whose last _A > 1_ blocks arrived late, even if that chain tip
+represents the "best" fork, since this would be the expected behavior of a selfish miner.
+
+* Since all nodes know about all burn transactions, the long tail of small burners
+(i.e. users who support leaders) can collectively throw their burns behind
+known-honest leaders' burns.  This increases the chance that honest leaders will
+be elected, thereby increasing the fraction of honest burn power and making it
+harder for a selfish leader to get elected.
+
+* The Stacks chain reward system spreads out rewards for creating
+blocks and mining transactions across a large interval.  This "smooths over"
+short-lived selfish mining attacks -- while selfish leaders still receive more
+than their fair share of the rewards, the lower variance imposed by the
+reward window makes this discrepancy smaller.
+
+* All Stacks nodes relay all blocks that correspond to on-chain commitments,
+even if they suspect that they came from the attacker.  If an honest leader finds two chain tips of equal
+length, it selects at random which chain tip to build off of.  This ensures that
+the fraction of honest burns that build on top of the attack fork versus the honest fork
+is statistically capped at 50% when they are the same length.
+
+None of these points _prevent_ selfish mining, but they give honest users and
+honest leaders the tools to make selfish mining more difficult to pull off than in
+PoW chains.  Depending on user activity, they also make economically-motivated
+leaders less likely to participate in a selfish miner cartel -- doing so always produces evidence,
+which honest leaders and users can act on to reduce or eliminate 
+their expected rewards.
+
+Nevertheless, these arguments are only intuitions at this time.  A more
+rigorous analysis is needed to see exactly how these points affect the
+profitibility of selfish mining.  Because the Stacks blockchain represents a
+clean slate blockchain design, we have an opportunity to consider the past
+several years of research into attacks and defenses against block-hiding
+attacks.  This section will be updated as our understanding evolves.
+
+# Fork Selection
+
+Fork selection in the Stacks blockchain requires a metric to determine which
+chain, between two candidates, is the "best" chain.  Using proof-of-burn as the
+security method for the blockchain implies a direct metric:  the total sum of
+burns in the election blocks for a candidate chain.  In particular, **the Stacks
+blockchain measures a fork's quality by the total amount of burns which _confirms_ block _N_** (as
+opposed to the amount of burn required for the _election_ of block _N_).
```

Comment:
I don't think the above paragraph matches a "longest valid fork" rule --- my understanding of the longest valid fork rule was to simply count epochs, regardless of how much was burned in any particular epoch. That's why we wanted to have a minimal burn amount for epochs.

Reply:
Yeah, that's a typo.  We want "longest valid fork" not "fork with the most burns."